modules/azure/cosmosdb.go

```diff
@@ -0,0 +1,197 @@
+	if err != nil {
+		return nil, err
+	}
+
+	// Attach authorizer to the client
+	cosmosClient.Authorizer = *authorizer
+
+	return cosmosClient, nil
+}
+
+// GetCosmosDBSQLClient is a helper function that will setup a CosmosDB SQL client. This function would fail the test if there is an error.
+func GetCosmosDBSQLClient(t testing.TestingT, subscriptionID string) *documentdb.SQLResourcesClient {
+	cosmosClient, err := GetCosmosDBSQLClientE(subscriptionID)
+	require.NoError(t, err)
+
+	return cosmosClient
+}
+
+// GetCosmosDBSQLDatabase is a helper function that gets a SQL database. This function would fail the test if there is an error.
+func GetCosmosDBSQLDatabase(t testing.TestingT, subscriptionID string, resourceGroupName string, accountName string, databaseName string) *documentdb.SQLDatabaseGetResults {
+	cosmosSQLDB, err := GetCosmosDBSQLDatabaseE(t, subscriptionID, resourceGroupName, accountName, databaseName)
+	require.NoError(t, err)
+
+	return cosmosSQLDB
+}
+
+// GetCosmosDBSQLDatabaseE is a helper function that gets a SQL database.
+func GetCosmosDBSQLDatabaseE(t testing.TestingT, subscriptionID string, resourceGroupName string, accountName string, databaseName string) (*documentdb.SQLDatabaseGetResults, error) {
+	// Create a CosmosDB client
+	cosmosClient, err := GetCosmosDBSQLClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the corresponding database account
+	cosmosSQLDB, err := cosmosClient.GetSQLDatabase(context.Background(), resourceGroupName, accountName, databaseName)
+	if err != nil {
+		return nil, err
+	}
+
+	//Return DB
+	return &cosmosSQLDB, nil
+}
+
+// GetCosmosDBSQLContainer is a helper function that gets a SQL container. This function would fail the test if there is an error.
+func GetCosmosDBSQLContainer(t testing.TestingT, subscriptionID string, resourceGroupName string, accountName string, databaseName string, containerName string) *documentdb.SQLContainerGetResults {
+	cosmosSQLContainer, err := GetCosmosDBSQLContainerE(t, subscriptionID, resourceGroupName, accountName, databaseName, containerName)
+	require.NoError(t, err)
+
+	return cosmosSQLContainer
+}
+
+// GetCosmosDBSQLContainerE is a helper function that gets a SQL container.
+func GetCosmosDBSQLContainerE(t testing.TestingT, subscriptionID string, resourceGroupName string, accountName string, databaseName string, containerName string) (*documentdb.SQLContainerGetResults, error) {
+	// Create a CosmosDB client
+	cosmosClient, err := GetCosmosDBSQLClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the corresponding database account
+	cosmosSQLContainer, err := cosmosClient.GetSQLContainer(context.Background(), resourceGroupName, accountName, databaseName, containerName)
+	if err != nil {
+		return nil, err
+	}
+
+	//Return DB
+	return &cosmosSQLContainer, nil
+}
+
+// GetCosmosDBSQLDatabaseThroughput is a helper function that gets a SQL database. This function would fail the test if there is an error.
```

Comment:
It looks like a lot of these comments need to be updated. E.g., this function is returning the DB throughput, not the database itself, right?

Reply:
Thanks for your feedback! I have updated the comments.